qutebrowser/completion/completer.py

```diff
@@ -151,6 +151,19 @@ def _get_completion_model(self, completion, parts, cursor_part):
             except KeyError:
                 # No completion model for this section/option.
                 model = None
+        elif completion == usertypes.Completion.keybinding:
+            # look for --mode to provide mode-specific binding completions
+            # since flags are ignored by most completers, we need to re-parse
+            parts, cursor_part = self._filter_cmdline_parts(self.split(),
+                                                            cursor_part,
+                                                            ignore_flags=False)
+            if parts[cursor_part - 1].startswith("--mode="):
```

Comment:
The `Command` has that information as `AnnotationInfo.kwargs['nargs']` - but I really think the whole completion code needs a good refactoring at some point with tests and those usecases in mind... Currently it's way too hard to write a completion for something, and I'm not sure if piling more and more complexity on it at the moment is a good idea :worried: